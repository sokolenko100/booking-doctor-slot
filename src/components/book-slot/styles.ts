import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  bookSlotButton: {
    width: '100%',
    flexDirection: 'row',
    alignItems: 'center',
    padding: 10,
    backgroundColor: '#f8f9fa',
    flex: 1,
    height: 45,
    borderRadius: 5,
    borderColor: '#bcbcbc',
    borderWidth: 0.2,
  },
  bookSlotText: {
    color: '#007BFF',
    fontSize: 14,
    marginLeft: 5,
    fontWeight: '600',
  },
});
