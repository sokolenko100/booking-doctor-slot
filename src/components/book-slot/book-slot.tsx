import React, {FC, memo} from 'react';
import {Text, TouchableOpacity} from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';
import {ITimeSlot} from '../../interfaces/interface';
import {styles} from './styles';

interface BookSlotProps {
  slot: ITimeSlot;
  onSlotPress: (slot: ITimeSlot) => void;
}

/**
 * BookSlot component for booking a time slot.
 * @param {ITimeSlot} slot - The time slot object containing information about the slot.
 * @param {(slot: ITimeSlot) => void} onSlotPress - A callback function to handle the slot press event.
 * @returns {JSX.Element} - A TouchableOpacity component that, when pressed, triggers the onSlotPress function with the provided slot object.
 */
export const BookSlot: FC<BookSlotProps> = memo(({slot, onSlotPress}) => {
  return (
    <TouchableOpacity
      testID="book-slot-buttonID"
      onPress={() => onSlotPress(slot)}
      style={styles.bookSlotButton}>
      <Icon name="add" size={20} color="#007BFF" />
      <Text testID="book-slot-textID" style={styles.bookSlotText}>
        Book slot
      </Text>
    </TouchableOpacity>
  );
});
