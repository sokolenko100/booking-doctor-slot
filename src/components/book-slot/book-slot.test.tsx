import {fireEvent, render} from '@testing-library/react-native';
import React from 'react';
import {BookSlot} from './book-slot';

const mockSlot = {
  id: '1',
  Start: '09:00',
  End: '10:00',
  textColor: 'blue',
};

const mockOnSlotPress = jest.fn();

describe('BookSlot component', () => {
  it('renders with text Book slote', () => {
    const {getByText} = render(
      <BookSlot slot={mockSlot} onSlotPress={mockOnSlotPress} />,
    );

    expect(getByText('Book slot')).toBeTruthy();
  });

  it('calls onSlotPress function when TouchableOpacity is pressed', () => {
    const {getByTestId} = render(
      <BookSlot slot={mockSlot} onSlotPress={mockOnSlotPress} />,
    );

    fireEvent.press(getByTestId('book-slot-buttonID'));
    expect(mockOnSlotPress).toHaveBeenCalledWith(mockSlot);
  });
  it('matches the snapshot', async () => {
    const output = render(
      <BookSlot slot={mockSlot} onSlotPress={mockOnSlotPress} />,
    );

    expect(output).toMatchSnapshot();
  });
});
