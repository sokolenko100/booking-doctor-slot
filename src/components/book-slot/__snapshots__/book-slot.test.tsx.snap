// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookSlot component matches the snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#f8f9fa",
      "borderColor": "#bcbcbc",
      "borderRadius": 5,
      "borderWidth": 0.2,
      "flex": 1,
      "flexDirection": "row",
      "height": 45,
      "opacity": 1,
      "padding": 10,
      "width": "100%",
    }
  }
  testID="book-slot-buttonID"
>
  <View
    style={
      {
        "backgroundColor": "#007BFF",
        "height": 20,
        "width": 20,
      }
    }
  />
  <Text
    style={
      {
        "color": "#007BFF",
        "fontSize": 14,
        "fontWeight": "600",
        "marginLeft": 5,
      }
    }
    testID="book-slot-textID"
  >
    Book slot
  </Text>
</View>
`;
